tests/test_merge_db_functions.py

```diff
@@ -0,0 +1,481 @@
+      },
+      '1-19-19-1x1-64-19-19-1024-0x0-1x1-1x1-0-NCHW-FP32-W': {
+          'miopenConvolutionBwdWeightsAlgoDirect':
+              'ConvAsmBwdWrW1x1,0.445761,0,miopenConvolutionBwdWeightsAlgoDirect,not used',
+          'miopenConvolutionBwdWeightsAlgoWinograd':
+              'ConvBinWinogradRxSf2x3,11.9268,0,miopenConvolutionBwdWeightsAlgoWinograd,not used',
+          'miopenConvolutionBwdWeightsAlgoImplicitGEMM':
+              'ConvMlirIgemmWrWXdlops,69.8604,0,miopenConvolutionBwdWeightsAlgoImplicitGEMM,not used'
+      },
+      '1024-14-14-1x1-256-14-14-15-0x0-1x1-1x1-0-NCHW-FP16-F': {
+          'miopenConvolutionFwdAlgoImplicitGEMM':
+              'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.03808,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+          'miopenConvolutionFwdAlgoDirect':
+              'ConvAsm1x1U,0.11072,0,miopenConvolutionFwdAlgoDirect,not used',
+          'miopenConvolutionFwdAlgoWinograd':
+              'ConvBinWinogradRxSf3x2,0.208159,0,miopenConvolutionFwdAlgoWinograd,not used',
+          'miopenConvolutionFwdAlgoGEMM':
+              'GemmFwd1x1_0_1,2730.97,0,miopenConvolutionFwdAlgoGEMM,not used'
+      }
+  })
+
+
+def test_write_merge_results():
+
+  master_list = {
+      '1-48-480-3x3-16-48-480-1-1x1-1x1-1x1-0-NCHW-FP16-F': {
+          'miopenConvolutionFwdAlgoImplicitGEMM':
+              'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02512,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+          'miopenConvolutionFwdAlgoDirect':
+              'ConvOclDirectFwd,0.02544,0,miopenConvolutionFwdAlgoDirect,not used',
+          'miopenConvolutionFwdAlgoWinograd':
+              'ConvBinWinogradRxSf2x3g1,0.03104,0,miopenConvolutionFwdAlgoWinograd,not used',
+          'miopenConvolutionFwdAlgoGEMM':
+              'GemmFwdRest,0.06096,414720,miopenConvolutionFwdAlgoGEMM,not used'
+      }
+  }
+
+  final_file = '/data/tests/final.txt'
+  copy_files = []
+
+  key = '1-48-480-3x3-16-48-480-1-1x1-1x1-1x1-0-NCHW-FP16-F'
+  write_merge_results(master_list, final_file, copy_files)
+
+  if key == open('/data/tests/final.txt').read():
+    assert (True)
+    f.close()
+
+  #input set to invalid types
+  master_list = {}
+
+  final_file = '/data/tests/final.txt'
+  key = {0}
+  write_merge_results(master_list, final_file, copy_files)
+
+  if key == open('/data/tests/final.txt').read():
+    assert (True)
+    f.close()
+
+
+def test_merge_text_file():
+  master_file = '/data/utils/test_files/old_gfx90a68.HIP.fdb.txt'
+  copy_only = False
+  keep_keys = False
+  target_file = '/data/utils/test_files/usr_gfx90a68.HIP.fdb.txt'
+
+  result_file = merge_text_file(master_file, copy_only, keep_keys, target_file)
+
+  if (os.stat(result_file)):
+    assert (True)
+  else:
+    assert (False)
+
+  try:
+    assert [row for row in open(master_file)
+           ] == [row for row in open(target_file)]
+  except AssertionError as msg:
+    assert (True)
```

Comment:
if master == target then the assert is true and this still passes
correct / simplify to
```py
assert statement == False
```